examples/js/loaders/GLTFLoader.js

```diff
@@ -2080,7 +2080,7 @@ THREE.GLTFLoader = ( function () {
 
 						}
 
-						if ( primitive.extras ) mesh.userData = primitive.extras;
+						Object.assign(mesh.userData, JSON.parse(JSON.stringify(meshDef.extras)), JSON.parse(JSON.stringify(primitive.extras)));
```

Comment:
I think it's fine to reference, without deep-copying, the `extras` object. That's what we've done with the other `.extras` properties during parsing.

Also, let's use a loop here instead of `Object.assign`, despite my comment earlier. IE11 does not support `Object.assign`.

```js
if ( meshDef.extras !== undefined ) mesh.userData = meshDef.extras;

if ( primitive.extras !== undefined ) {

  mesh.userData = mesh.userData || {};

  for ( var key in primitive.extras ) {

    if ( mesh.userData[ key ] === undefined ) mesh.userData[ key ] = primitive.extras[ key ];

  }
}
```

Reply:
Okay, will copy by reference then.
Ah... IE11... forgot about this stupid thing :/ yeah we should go with the loop. Just tested it and it seems to work fine.
Any reason for ```if ( primitive.extras !== undefined )``` instead of ```if ( primitive.extras )``` ?

Reply:
Also there is the question if the ```primitive.extras```should overwrite the ```mesh.extras``` or not? 
In the case you posted you don't overwrite them, but they MAY be more important. I don't really like it so much that information may get lost, but for now I will implement this case.